static/app/views/starfish/components/detailPanel.tsx

```diff
@@ -32,13 +32,17 @@ export default function Detail({children, detailKey, onClose}: DetailProps) {
 
   const panelRef = useRef<HTMLDivElement>(null);
   useOnClickOutside(panelRef, () => {
-    onClose?.();
+    if (!state.collapsed) {
+      onClose?.();
+    }
     setState({collapsed: true});
```

Comment:
Let's put the `setState` calls in the condition as well so we don't cause an unnecessary rerender
```suggestion
    if (!state.collapsed) {
      onClose?.();
      setState({collapsed: true});
    }
```